core/src/components/datetime/utils/data.ts

```diff
@@ -254,7 +254,22 @@ export const generateTime = (
  * Given DatetimeParts, generate the previous,
  * current, and and next months.
  */
-export const generateMonths = (refParts: DatetimeParts): DatetimeParts[] => {
+export const generateMonths = (refParts: DatetimeParts, forcedDate: DatetimeParts | null = null): DatetimeParts[] => {
+  /**
+   * If we're forcing a month to appear, and it's different from the current month,
+   * ensure it appears by replacing the next or previous month as appropriate.
+   */
+  if (forcedDate !== null && (refParts.month !== forcedDate.month || refParts.year !== forcedDate.year)) {
+    const current = { month: refParts.month, year: refParts.year, day: refParts.day };
```

Comment:
Nit: You could pull `current` outside of the `if` block and use it in both return statements.